### PR TITLE
NLopt: remove internal find, check and build; use find_package with namespace

### PR DESCRIPTION
**Pull request by pierrepebay**
### What I did to build nlopt:
```
# from MUQ root:
mkdir deps
mkdir deps/nlopt
cd deps/nlopt
mkdir build install
# we can change this to .tar if needed
wget https://github.com/stevengj/nlopt/archive/refs/tags/v2.8.0.zip
unzip v2.8.0
cmake -S nlopt-2.8.0/ -B build/ -DCMAKE_INSTALL_PREFIX=~/Develop/MIT-MUQ/deps/nlopt/install/
cd build
make -j24 && make install
```

### How I used this build with MUQ:
```
cmake -DCMAKE_C_COMPILER=gcc \
      -DCMAKE_CXX_COMPILER=g++ \
      -S /home/plpebay/Develop/MIT-MUQ/ \
      -B /home/plpebay/Develop/MIT-MUQ/build/ \
      -DMUQ_USE_GTEST=ON \
      -DMUQ_GTEST_DIR=/home/plpebay/Develop/MIT-MUQ/deps/googletest/install \
      -DCMAKE_INSTALL_PREFIX=/home/plpebay/Develop/MIT-MUQ/install \
      -DNLopt_DIR=/home/plpebay/Develop/MIT-MUQ/deps/nlopt/install/lib/cmake/nlopt \
      -DHDF5_DIR=/home/plpebay/Develop/MIT-MUQ/deps/hdf5/install/share/cmake
make -j24 && make install
```

### CODE REVIEW

**Comment by pierrepebay** [comment]
The version used for the internal build of NLopt is 2.4.2, which doesn't use CMake for the build process (uses a `configure`); so I use the most recent version of NLopt, which builds and passes tests with MUQ.

**Comment by fnrizzi** [comment]
> The version used for the internal build of NLopt is 2.4.2, which doesn't use CMake for the build process (uses a `configure`); so I use the most recent version of NLopt, which builds and passes tests with MUQ.

what version did you use?

File: cmake/finds/RequiredDependencySearch.cmake [hunk 1]
```diff
@@ -79,6 +78,17 @@ if(MUQ_USE_OPENMPI)
...
+############################
+##### LOOK FOR NLOPT  ######
+############################
+
+set(HAVE_NLopt 1)
+
+find_package(NLopt REQUIRED)
+if (NLopt_FOUND)
+    set(MUQ_HAS_NLopt 1)
+endif ()
```
Comment (fnrizzi) [inline cmake/finds/RequiredDependencySearch.cmake:90]: i don't understand this: if this is required how can the if afterward be any useful?

File: cmake/finds/RequiredDependencySearch.cmake [hunk 2]
```diff
@@ -79,6 +78,17 @@ if(MUQ_USE_OPENMPI)
 
 endif()
 
+############################
+##### LOOK FOR NLOPT  ######
+############################
+
+set(HAVE_NLopt 1)
```
Comment (fnrizzi) [inline cmake/finds/RequiredDependencySearch.cmake:85]: what is this used for?

File: cmake/finds/RequiredDependencySearch.cmake (hunk 1 above)
Comment (fnrizzi) [reply cmake/finds/RequiredDependencySearch.cmake:90]: please address as discussed

File: cmake/finds/RequiredDependencySearch.cmake (hunk 2 above)
Comment (fnrizzi) [reply cmake/finds/RequiredDependencySearch.cmake:85]: please address as discussed

**Comment by fnrizzi** [comment]
please also resolve conflicts

**Comment by pierrepebay** [comment]
Rebased; updated the top comment for build instructions as we now need to point to both HDF5 and NLopt

**Pull request merged**